osu.Game/Screens/Edit/List/DrawableList.cs

```diff
@@ -0,0 +1,146 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System;
+using System.Collections.Generic;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Containers;
+using osu.Framework.Input.Events;
+using osu.Game.Graphics.UserInterface;
+using osuTK;
+using osuTK.Input;
+
+namespace osu.Game.Screens.Edit.List
+{
+    public class DrawableList<T> : CompositeDrawable, IDrawableListItem<T>
```

Comment:
Is this something like `RearrangeableListContainer`? You might want to check whether you can already lean on an existing implementation like that.